File: 443-string-compression/443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:   
        s = ""
        l = r = 0
        n = len(chars)
        
        if n == 1:
            return n
    
        while r < n:
            
            while r < n and chars[r] == chars[l]:
                r += 1
            leng = r - l
            s += chars[l]
            if leng > 1:
                s += str(leng)
            l = r
        chars[:] = list(s)
        return len(chars)

File: 443-string-compression/443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int beg = 0, end = 0, n = chars.size();
        string res;
        while (end < n) {
            while (end < n && chars[beg] == chars[end])
                end++;
            int len = end - beg;
            res += chars[beg];
            if (len > 1)
                res += to_string(len);
            beg = end;
        }
        chars.clear();
        for (auto x: res)
            chars.push_back(x);
        return chars.size();
    }
};